qt/scientific_interfaces/ISISReflectometry/GUI/Runs/SearchResult.cpp

```diff
@@ -5,14 +5,51 @@
 //     & Institut Laue - Langevin
 // SPDX - License - Identifier: GPL - 3.0 +
 #include "SearchResult.h"
+#include <boost/regex.hpp>
 
 namespace MantidQt {
 namespace CustomInterfaces {
+
+SearchResult::SearchResult(const std::string &runNumber,
+                           const std::string &desc, const std::string &loc)
+    : m_runNumber(runNumber), m_description(desc), m_location(loc),
+      m_groupName(), m_theta() {
+  parseMetadata();
+}
+
+void SearchResult::parseMetadata() {
+  static boost::regex descriptionFormatRegex("(.*)(th[:=]([0-9.]+))(.*)");
+  boost::smatch matches;
+  if (boost::regex_search(m_description, matches, descriptionFormatRegex)) {
+    constexpr auto preThetaGroup = 1;
+    constexpr auto thetaValueGroup = 3;
+    m_theta = matches[thetaValueGroup].str();
+    m_groupName = matches[preThetaGroup].str();
+  } else {
+    m_groupName = m_description;
+  }
+}
+
+std::string SearchResult::runNumber() const { return m_runNumber; }
+
+std::string SearchResult::description() const { return m_description; }
+
+std::string SearchResult::location() const { return m_location; }
+
+std::string SearchResult::error() const { return m_error; }
+
+std::string SearchResult::groupName() const { return m_groupName; }
+
+std::string SearchResult::theta() const { return m_theta; }
+
```

Comment:
Should these 6 member function instead return `const std::string &`, which better matches the argument for ([QString::fromStdString]https://doc.qt.io/qt-5/qstring.html#fromStdString)?